### PR TITLE
[arch] #234 residual: arg-contract drift (02#6/07#0) + mock-audit hardening (12#0)

### LINKED ISSUE #289: [arch] #234 residual: arg-contract drift (02#6/07#0) + mock-audit hardening (12#0)
Follow-up to the arch-review 2026-06-12 (meta tracker #270). Issue #234 was auto-closed when its infra PRs merged, but 3 of its 5 findings were never addressed. This issue tracks fixing those 3. **Do not reopen #234.**

### 02#6 — single-tenant prompt emits a dead `pipeline=` arg
`apps/agents/graph/base.py` (single-tenant, no-data branch) instructs the agent to call `run_materialization` with `pipeline="<name>"`, but the MCP tool (`mcp_server/server.py`) has no `pipeline` parameter — routing moved into `materialize_workspace` per-provider, and all real params are injected server-side/hidden, so the LLM-facing schema has zero properties. The multi-tenant branch already omits the argument.
**Fix:** rewrite the single-tenant no-data prompt to stop emitting the dead arg; pin it with a test asserting `pipeline=` never appears in the schema-context text.

### 07#0 — context-injection contract relies on silent arg-dropping
The injecting node adds `workspace_id/user_id/thread_id/tool_call_id` to **every** MCP tool call, including the 7 read tools that declare only `workspace_id`. This works solely because LangChain's `BaseTool._parse_input` returns dict-schema input unvalidated **and** FastMCP's `ArgModelBase` defaults to `extra='ignore'`. If either library tightens, every MCP call fails at once.
**Fix:** add a guardrail test that pins the contract — inject the full arg set into a tool that declares none of the extras and assert they're accepted-and-ignored, so a future tightening fails loudly here, not in prod.

### 12#0 — mock-audit hardening
The suite mocks the exact seams that hid production bugs. Make it able to catch the known incident classes. In-scope seams (numbered from the original finding):
1. `reconcile_stale_thread_job`'s RUNNING false-failure branch is never built (every stale-reconcile test builds PENDING).
2. jobs_cancel tests patch the module-level `current_app` binding, so a surviving-sibling binding failure can't fire.
3. MCP tool tests set `tenant_schema.tenant_membership.tenant.external_id` — a path that does not exist on the real model (prod reads `tenant_schema.tenant.external_id`).
4. Resume test asserts `oauth_tokens` in agent config, codifying dead OAuth plumbing as contract.
6. `cancel_materialization` test asserts only `cancelled==True`, never the persisted state.
7. Stale-cursor regression injects `prior_run` via a mocked `filter/exclude/order_by` chain, so the real prior-run selection query is untested.
8. merge-users suite has zero `TenantMetadata` assertions — the repoint/cascade path is invisible.
9. OCS/Connect loader tests are happy-path MagicMock only (no 401/403/malformed/retry).

**Out of scope (owned elsewhere):** 12#0 item (5) sql_validator → #244; item (10) transformation trigger → #241.

This is test-hardening: each seam asserts behavior that passes against current production code (02#6 also fixes the prompt). No production redesign.

### PR DESCRIPTION
**Pull request by bderenzi**
Closes #289. Resolves the 3 unaddressed findings from arch-review issue #234 (auto-closed when its infra PRs merged — **not reopened**). Meta tracker: #270.

## What this does
One production fix (a dead prompt arg) plus a focused test-hardening pass that makes the suite able to catch the incident classes the existing mocks were hiding. Each seam asserts behavior that passes against current production code; no production redesign.

### 02#6 — drop dead `pipeline=` arg (production prompt fix)
`apps/agents/graph/base.py` single-tenant no-data branch told the agent to call `run_materialization` with `pipeline="<name>"`, but the MCP tool has **no** `pipeline` parameter (routing moved into `materialize_workspace` per-provider) and all real params are injected/hidden — the LLM-facing schema is empty. The multi-tenant branch already omitted it. Aligned the single-tenant branch and pinned it with a test asserting `pipeline=` never appears. (No `xfail` existed in the repo to remove.)

### 07#0 — pin the context-injection contract (guardrail)
The injecting node adds `workspace_id/user_id/thread_id/tool_call_id` to every MCP call, including the read tools that declare only `workspace_id`. This works solely because LangChain's `BaseTool._parse_input` passes dict-schema input unvalidated **and** FastMCP's generated `ArgModelBase` ignores unknown fields (`extra='ignore'`). New `tests/agents/test_context_injection_contract.py` pins both halves so a future tightening fails loudly here instead of prod-wide.

### 12#0 — mock-audit hardening
| # | Seam | Now |
|---|------|-----|
| 1 | `reconcile_stale_thread_job` RUNNING branch never built | stale-RUNNING → FAILED (interrupted summary) + fresh-RUNNING left alone (02#9 guard) |
| 2 | cancel tests patched the `current_app` binding, masking the surviving-sibling import | patch `JobManager.cancel_job_by_id_async` at the class level; the real binding must resolve for the abort to fire (+ a binding-is-live guard) |
| 3 | mock used non-existent `tenant_schema.tenant_membership.tenant.external_id` | set the real `tenant_schema.tenant.external_id`, delete the bogus path, assert `tenant_id` flows through |
| 4 | resume test pinned dead `oauth_tokens` plumbing | assert the real consumed contract (`configurable.thread_id`) |
| 6 | cancel test asserted only `cancelled==True` | assert persisted state (`FAILED`) + `asave` update_fields + `result.cancelled` |
| 7 | stale-cursor test mocked the `filter/exclude/order_by` chain | real-DB rows exercise `_load_prior_resume_cursors` end to end |
| 8 | merge-users had zero `TenantMetadata` assertions | repoint carries metadata across; conflict-delete keeps the canonical's |
| 9 | OCS/Connect loaders happy-path only | 403 / 5xx / malformed-JSON error paths |

**Out of scope (owned elsewhere, untouched):** 12#0 item 5 (sql_validator → #244), item 10 (transformation trigger → #241).

## Notes
- Item 6 pins the **current** behavior: `cancel_materialization` writes `RunState.FAILED` (not `CANCELLED`) while the cancel endpoint writes `CANCELLED`. That FAILED-vs-CANCELLED inconsistency is tracked separately in **#290** — this PR only makes it visible, it does not change it.
- Item 8 surfaced a minor reporting quirk: `report.tenant_membership_conflict_deleted` counts cascade-deleted rows too (so it's inflated when metadata exists). The tests assert the data outcome rather than that count.

## Verification
- `ruff check .` clean; `ruff format --check` clean.
- All touched modules + the async-conventions CI guard: **240 passed**.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: tests/test_jobs_endpoints.py [hunk 1]
```diff
@@ -786,3 +788,183 @@ async def test_active_jobs_does_not_reconcile_fresh_jobs():
...
+#
+# These tests leave the binding intact and patch the abort at the JobManager
+# CLASS level, so the real binding must resolve for the abort to fire.
+# ---------------------------------------------------------------------------
+
+
+def test_jobs_cancel_current_app_binding_is_live_not_a_blueprint():
+    """The surviving-sibling binding must resolve to a real App exposing a
+    usable job_manager — not procrastinate's not-ready FutureApp blueprint."""
+    from apps.workspaces.api import jobs_cancel
```
[the remark this answers is not in the document]
Comment (bderenzi) [reply tests/test_jobs_endpoints.py]: Fixed in b778259 — hoisted `from apps.workspaces.api import jobs_cancel` to the module-level import block and dropped the inline statement. Thanks!
